engfrosh_bot/__main__.py

```diff
@@ -148,6 +148,29 @@ async def discord_queue_callback(command: dict):
         logger.warning("Object type %s not supported. Ignoring command" % command["object"])
         return
 
+# Add Bot Listner for when someone gets a role
+intents = discord.Intents.default()
+intents.members = True
+intents.presences = True
+client = discord.Client(intents=intents)
+
+@client.event
+async def on_ready():
+    general_channel = client.get_channel(852503379430473801)
+    await general_channel.send('Bot Activated')
+
+@client.event
+async def on_member_join(member):
```

Comment:
This is excellent what you've added with this function, and it was going to be needed to add at some point anyway, however, the issue was to send a message when someone gets a role assigned, not when they join the server. Keep what you have here, but add another listener which uses member updates, [link](https://discordpy.readthedocs.io/en/stable/api.html#discord.on_member_update)